connector/src/main/java/org/jboss/as/connector/util/RaServicesFactory.java

```diff
@@ -89,6 +91,17 @@ public static void createDeploymentService(final ManagementResourceRegistration
                 .addDependency(NamingService.SERVICE_NAME)
                 .addDependency(ConnectorServices.BOOTSTRAP_CONTEXT_SERVICE.append(bootStrapCtxName))
                 .addDependency(ConnectorServices.RESOURCE_ADAPTER_DEPLOYER_SERVICE_PREFIX.append(connectorXmlDescriptor.getDeploymentName()));
+        String raName = deployment;
+        if (raxml.getId() != null) {
+            raName = raxml.getId();
+        }
+        ServiceName parentName = ServiceName.of(ConnectorServices.RA_SERVICE, raName);
+        for(ServiceName subServiceName: serviceRegistry.getServiceNames()) {
+            if (parentName.isParentOf(subServiceName)
```

Comment:
This return true if parentName.equals(subServiceName). Do you want the dependency for parentName as well?

This is due to the contract of ServiceName.isParentOf():

```
    /**
     * Determine whether this service name is the same as, or a parent of, the given service name.
     *
     * @param other the other name
     * @return {@code true} if this service name is a parent
     */
    public boolean isParentOf(ServiceName other) {
        return other != null && (equals(other) || isParentOf(other.parent));
    }
```